docs/extend/api.md

```diff
@@ -12,348 +12,967 @@ To use the built-in REST API as part of an integration, see [Consuming the REST
 
-return Tag::all()->load($relations);
+public function fields(): array
+{
+    return [
+        Schema\Str::make('name')
+            ->property('name_column'),
+    ];
+}
 ```
 
-### Extending API Controllers
+### Relationship aggregates
 
-It is possible to customize all of these options on _existing_ API controllers too via the `ApiController` extender
+The `Number` & `Integer` attributes are able to get efficiently get relationship aggregates such as counts, sums, max, and min.
 
 ```php
-use Flarum\Api\Event\WillGetData;
-use Flarum\Api\Controller\ListDiscussionsController;
-use Illuminate\Contracts\Events\Dispatcher;
+use Flarum\Api\Schema\Number;
+use Flarum\Api\Schema\Integer;
 
-return [
-    (new Extend\ApiController(ListDiscussionsController::class))
-        ->setSerializer(MyDiscussionSerializer::class)
-        ->addInclude('user')
-        ->addOptionalInclude('posts')
-        ->setLimit(20)
-        ->setMaxLimit(50)
-        ->setSort(['name' => 'asc'])
-        ->addSortField('firstName')
-        ->prepareDataQuery(function ($controller) {
-            // Add custom logic here to modify the controller
-            // before data queries are executed.
-        })
-]
+Integer::make('commentCount')
+    ->countRelation('comments'),
+
+Number::make('avgRevenue')
+    ->avgReation('reports', 'revenue'),
+
+Number::make('revenueSum')
+    ->sumRelation('reports', 'revenue'),
+
+Number::make('minNumber')
+    ->minRelation('posts', 'number'),
+
+Number::make('maxNumber')
+    ->maxRelation('posts', 'number'),
 ```
 
-The `ApiController` extender can also be used to adjust data before serialization
+## Relationships
+
+A relationship is a field, so all that was mentioned about fields above applies to relationships as well.
+
+There are two types of relationships: `ToOne` and `ToMany`. You can use the `Schema\Relationship\ToOne` and `Schema\Relationship\ToMany` classes to define them.
 
 ```php
-use Flarum\Api\Event\WillSerializeData;
-use Flarum\Api\Controller\ListDiscussionsController;
-use Illuminate\Contracts\Events\Dispatcher;
+use Flarum\Api\Schema;
 
-return [
-    (new Extend\ApiController(ListDiscussionsController::class))
-        ->prepareDataForSerialization(function ($controller, $data, $request, $document) {
-            $data->load('myCustomRelation');
-        }),
-]
+public function fields(): array
+{
+    return [
+        Schema\Relationship\ToOne::make('user')
+            ->type('users'),
+        Schema\Relationship\ToMany::make('children')
+            ->type('labels'),
+    ];
+}
 ```
 
-## Serializers
+### Inclusion & Linkage
+
+You can mark a relationship as includable through the `includable` method. This means that the relationship can be included in the API response. You can also use the `withLinkage` and `withoutLinkage` methods to determine whether the relationship ID(s) should be included in the API response (`ToMany` relationships are not linked by default contrary to `ToOne` relationships).
```

Comment:
When would you have non-includable relationships?

Reply:
When you only need a relationship to be writable perhaps. Though not a common use case, it remains valid.

Reply:
Might be worth adding a short tip about this